graspologic/utils/utils.py

```diff
@@ -223,7 +224,17 @@ def is_unweighted(
 
 
 def is_almost_symmetric(X, atol=1e-15):
-    return abs(X - X.T).max() <= atol
+    if (X.ndim != 2) or (X.shape[0] != X.shape[1]):
+        return False
+    if isinstance(X, (np.ndarray, scipy.sparse.spmatrix)):
+        return abs(X - X.T).max() <= atol
+    elif isinstance(X, LinearOperator):
+        n, _ = X.shape
+        u = np.random.rand(n)
+        v = np.random.rand(n)
+        return np.allclose(u.T @ (X @ v), v.T @ (X @ u))
```

Comment:
Modified this function to allow for LinearOperator objects and to do a dimensionality check.
sklearn also has a [check_symmetric](https://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.check_symmetric.html) function, but doesn't return a boolean -- maybe worth checking out the code to see if there's anything we need to steal

Reply:
Yeah - it felt icky to me too. I chose to do a randomized algorithm because the whole point of using LinearOperator is to avoid doing computations on large matrices, and I think you'd necessarily have to do a computation on a large matrix if you did the full check.

I think the way I'd do the full check is to do $A e_i - A ^\top e_i$  for $e_i$ the columns of the identity matrix, if we wanted to bite the bullet and avoid the randomized algorithm, it'd just be way, way slower on huge matrices (although I could do some profiling to check that). 

I'm not sure if scipy has a check for this, I scanned briefly last time and didn't see anything. I can look again more deeply though.

Reply:
yeah just wondering if we really need this functionality at all

side note - does your casc implementation only work for undirected graphs?

Reply:
Yeah, I have a TODO to add support for directed, but I was thinking I'd add it in a future PR. I have to think through what I'd want it to look like. I guess you'd still sum `L+aYY^T`, with L directed now, and then you'd essentially do ASE, using the right singular vector for the 'out' latent positions?

Reply:
wait, can't we just check if the input graph is directed, for the current conversation? 

if we need it for the linear operator some other time, then yeah i suggest we save this conversation for then

Reply:
That would work for CASE, but wouldn't work for outside of CASE. The place where this check happens is [https://github.com/microsoft/graspologic/pull/599/files#diff-9f086a43899b53f980b51393b3c7d86df13c85692de0fb5eb2ae14fdc577b786R277](here), for the `eigsh` option in SelectSVD. `eigsh` is meant for symmetric matrices, but for some reason still works on non-symmetric matrices. 

I can do it like this if we want though. I could also just remove `eigsh` and use `sparse.svd`, even though it's slower, for the extra simplicity.

Reply:
lets at least do it for CASE. For selectSVD im kinda leaning towards just not having that check for now which does also feel a little icky but... scipy did it also? what do you think?

Reply:
> what did you end up doing regarding our conversation about checking for directed graphs?

Removed the check, put a symmetry check into `fit` prior to the LinearOperator conversion.
Had to add a `directed=None` kwarg to `_reduce_dim` in `BaseSpectralEmbed`, because it checks symmetry there and wouldn't have worked for a LinearOperator.

65db9d1